### PR TITLE
Prevent organizations being created with just integers in their slug

### LINKED ISSUE #20264: Prevent organizations being created with just integers in their slug
## Is this feature related to a problem? Please describe.

As we have introduced new routes to allow organizations to be queried in the API via either their slug or their ID, we need to prevent organizations from being created with just integers to avoid any unintentional collisions.

## Describe the solution you’d like

- Prevent organizations with usernames that are just integers

## Definition of Done

- [ ] When users created a new organization, they should not be able to enter in an organization username that is just integers (0 - 9). If this occurs, an error message should show, asking the user to add other characters to their username.

## Additional Context

- This is a scoped issue for organizations, based on the larger issue - https://github.com/forem/forem/issues/19758

### PR DESCRIPTION
**Pull request by danielmbrasil**
## What type of PR is this? (check all applicable)

- [ ] Refactor
- [X] Feature
- [ ] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

## Description
Add a validation to check if the slug/username contains at least one non-digit character. Add translations for the error message and tests.

## Related Tickets & Documents

- Related Issue #20264
- Closes #20264

## QA Instructions, Screenshots, Recordings

1. Try to create an organization with a slug containing only digits
2. Get an error message telling organizations' slugs must contain at least one non-digit character

No changes to the UI but here's a screenshot with the error message:
image

### UI accessibility checklist
_If your PR includes UI changes, please utilize this checklist:_
- [ ] Semantic HTML implemented?
- [ ] Keyboard operability supported?
- [ ] Checked with [axe DevTools](https://www.deque.com/axe/) and addressed `Critical` and `Serious` issues?
- [ ] Color contrast tested?

_For more info, check out the
[Forem Accessibility Docs](https://developers.forem.com/frontend/accessibility)._

## Added/updated tests?
- [X] Yes

### CODE REVIEW

File: config/locales/models/fr.yml [hunk 1]
```diff
@@ -79,6 +79,7 @@ fr:
     organization:
       integer_only: Entier seulement. Aucun signe n'est autorisé.
       reserved_word: "%<value>s est un mot réservé. Contactez l'administrateur du site pour obtenir de l'aide pour enregistrer votre organisation."
+      slug_format: doit inclure au moins un caractère non numérique. Veuillez ajouter d'autres caractères au nom d'utilisateur de votre organisation
```
Comment (danielmbrasil) [inline config/locales/models/fr.yml:82]: I used a translator for this so I'm not sure if this is an accurate translation 😅.

**Comment by danielmbrasil** [comment]
The last commit should fix the failed CI checks related to RSpec. However, the Crypress checks were also failing and I think it's unrelated. I have another open PR with similar fails. Am I missing something to get the Crypress CI passing?

**Review by lightalloy — CHANGES_REQUESTED** [review]
Thanks for the pr @danielmbrasil :raised_hands: 

Currently we have slug validations in `CrossModelSlugValidator` (and it already has format validation, but w/o restricting number-only slugs), so it would make sense to validate slug there.

Ideally, we should validate the same way for users, podcast episodes, etc, as mentioned in #19758

As I see, currently we only plan to validate orgs, but it still makes sense to keep slug validations in one place.

Also, there are organizations that have number-only slugs which should be updated before deploying the changes. It was supposed to be done manually by @benhalpern

**Comment by danielmbrasil** [comment]
Hey @lightalloy,

I've moved the validation to `CrossModelSlugValidator`. However,

> As I see, currently we only plan to validate orgs, but it still makes sense to keep slug validations in one place.

my latest change (be4f9e1c494e152472714028ea9fff973bf73e3b) should affect the slug format validations for the other models too. Should I use a second regex and validate for organizations only as a special case?

**Comment by benhalpern** [comment]
@lightalloy support task has been completed

**Comment by lightalloy** [comment]
@danielmbrasil yeah, we need a separate regex for organizations. 
We can't update the regex for other models (like users) at the moment because there are records with number-only slugs in the corresponding tables.

Let's check organizations with a separate regexp. That won't be super "beautiful", but I would accept that trade-off for now.

**Comment by danielmbrasil** [comment]
@lightalloy I just updated the PR to include the requested changes!

**Review by benhalpern — APPROVED** [review]
Looks good to me, but would love a final @lightalloy check

**Review by lightalloy — APPROVED** [review]
Looks good to me :raised_hands:

**Pull request merged**